GitVersionCore/BranchConfigurationCalculator.cs

```diff
@@ -117,8 +131,8 @@ public class BranchConfigurationCalculator
             else
                 errorMessage = "Failed to inherit Increment branch configuration, ended up with: " + string.Join(", ", possibleParents.Select(p => p.Name));
 
-            var hasDevelop = repository.Branches["develop"] != null;
-            var branchName = hasDevelop ? "develop" : "master";
+            var developBranch = repository.Branches.FirstOrDefault(b => Regex.IsMatch(b.Name, "^develop", RegexOptions.IgnoreCase));
```

Comment:
This can't be removed at the moment. If the code gets to the point where it cannot figure out which branch to inherit from then this code needs to be able to pick between master and develop.

An example:

```
* [master, develop]
|
* [feature/foo]
```

In this scenario there is no way to know if GitVersion should inheirit the config from master or develop, this code makes sure that develop gets selected. It isn't great but I can't think of a better solution?